pkg/api/frontend_logging.go

```diff
@@ -46,3 +47,35 @@ func NewFrontendLogMessageHandler(store *frontendlogging.SourceMapStore) fronten
 		return response.Success("ok")
 	}
 }
+
+func GrafanaJavascriptAgentLogMessageHandler() frontendLogMessageHandler {
+	return func(c *models.ReqContext) response.Response {
+		event := frontendlogging.FrontendGrafanaJavascriptAgentEvent{}
+		if err := web.Bind(c.Req, &event); err != nil {
+			return response.Error(http.StatusBadRequest, "bad request data", err)
+		}
+		var ctx = frontendlogging.CtxVector{}
+		ctx = event.AddMetaToContext(ctx)
+
+		if event.Logs != nil && len(event.Logs) > 0 {
```

Comment:
would be better to iterate over the slice rather than pick first item. on account that we start batching logs at some point :)

Reply:
good point, iterating now

Reply:
https://github.com/grafana/grafana/pull/50801/commits/ce40cfdc5e992e107e60e8453ff365b8ac8d9ea8